ciphers/hill_cipher.py

```diff
@@ -63,14 +64,34 @@ class HillCipher:
     def __init__(self, encrypt_key):
         """
         encrypt_key is an NxN numpy matrix
+        >>> HillCipher(numpy.matrix([[2,5],[1,6]])).__init__(numpy.matrix([[2,5],[1,6]]))
+        >>>
         """
         self.encrypt_key = self.modulus(encrypt_key)  # mod36 calc's on the encrypt key
         self.check_determinant()  # validate the determinant of the encryption key
         self.decrypt_key = None
         self.break_key = encrypt_key.shape[0]
 
+    def replace_letters(self, letter):
```

Comment:
Need type hints.  This function does not really _replace_ anything.  Can we rename to something more self documenting?  get_letter_by_letter or lookup_letter_by_letter.  Alternatively could we have a single function instead of two functions?  get_letter or lookup_letter could start with `isinstance(key, int)` and return the right result if `key` was a str or int.